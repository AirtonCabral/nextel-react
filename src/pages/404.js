import React from 'react'
import ConnectedLink from '../lib/link'

export const NotFound = () => (
    <div style={{ 'paddingLeft': 50 }}>
        <h1>PÃ¡gina nÃ£o encontrada :(</h1>
        <h2><span role="img">ğŸš‘ </span>Volte para o seu <ConnectedLink to="/">dashboard</ConnectedLink></h2>
    </div>
)

export default NotFound
