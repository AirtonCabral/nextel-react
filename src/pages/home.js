import React from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { startConnection } from '../actions/a_auth'
import { signIn, addToPortfolio, removeToPortfolio } from '../actions/a_user'
import { getProducts } from '../actions/a_portfolio'
import './../sass/home.scss'

import Modal from '@material-ui/core/Modal';

import ConnectionStatus from './../components/connection_status';
import MenuAppBar from './../components/menu_app_bar';
import TabContainer from './../components/tabs_app';
import CardsProducts from './../components/cards_products'
import Footer from './../components/footer';

import WelcomeModal from './../components/welcome';
import Details from './../components/details';
import NewsModal from './../components/news_modal';
import ConfirmModal from './../components/confirm_modal';

function rand() {
    return Math.round(Math.random() * 20) - 10;
}

function getModalStyle() {
    const top = 50 + rand();
    const left = 50 + rand();

    return {
        top: `50%`,
        left: `50%`,
        transform: `translate(-${top}%, -${left}%)`,
    };
}

var msisdn_received = '5521981981981';

export class Home extends React.Component {
    constructor(props) {
        super(props);

        // let msisdn = null;
        if (props.location.search) {
            let value = props.location.search;
            msisdn_received = value.split('=')[1];
        }

        this.state = {
            //consoleToggle: false,
            modalToggle: true,
            typeContent: 0, //0 = modal inicial, 1 = detalhes, 2 = modal de News, 3= 
            modalData: [],
            ready: false,
            messages: 'Iniciando',
            errors: null,
            // count: 27,
            // limitCount: 70,
            // services: [],
            // bleh:[],
            // msisdn
        };
    }

    componentDidMount() {

        let status_message = '';

        if (msisdn_received) {
            status_message = 'Autenticando...';
        }
        else {
            status_message = 'MSISDN Não encontrado :(';
            this.setState({
                messages: status_message,
                errors: true
            });
            return;
        }

        this.setState({ messages: status_message }, () => {
            this.props.startConnection('drweb', 'c62J3rZovtw', msisdn_received) // >>>>> auth
                .then(() => {
                    if (this.props.token && this.props.msisdn) {
                        this.props.signIn(this.props.token, this.props.msisdn)// >>>>> signin
                            .then(() => {
                                if (this.props.assinantesID) {
                                    this.setState({
                                        messages: 'Carregando Lista de Serviços',
                                    }, () => {
                                        this.props.getProducts(this.props.token)// >>>>> get products
                                            .then(() => {
                                                let message = ''
                                                if (this.props.products.length > 0) {
                                                    message = 'Lista Carregada!'
                                                    // START PORTFOLIO DEFAULT
                                                    this.props.sva_produtos_id.map((v, i) => {
                                                        this.props.products.map((_v, _i) => {
                                                            if (_v.id === v) {
                                                                this.props.addToPortfolio({ ..._v });
                                                            }
                                                        });
                                                    });
                                                    this.setState({ messages: message }, () => {
                                                        // START PROJECT
                                                        setTimeout(() => {
                                                            this.setState({
                                                                ready: true
                                                            });
                                                        }, 1000);
                                                    });
                                                }
                                                else {
                                                    message = 'Erro ao carregar lista de serviços'
                                                    // ERROR
                                                    this.setState({
                                                        messages: message,
                                                        errors: true
                                                    });
                                                }
                                            })
                                    });
                                }
                                else {
                                    this.setState({
                                        messages: 'Erro ao carregar usuário.',
                                        errors: true
                                    });
                                }
                            })
                    }
                    else {
                        this.setState({
                            messages: 'Problemas com autenticação :(',
                            errors: true
                        });
                    }
                })
        });
    }

    handleOpen = () => {
        this.setState({ 
            modalToggle: true,
            typeContent: 0
        });
    };

    handleClose = () => {
        this.setState({ modalToggle: false });
    };

    openDetails = (e, data) => {
        this.setState({
            modalToggle: true,
            modalData: data,
            typeContent: 1
        });
        console.log(this.state.modalData);
    };

    closeDetails = () => {
        this.setState({ modalToggle: false });
    };

    handleSwitch = (checked, value) => {
        if (checked) {
            this.props.addToPortfolio(value);
        }
        else {
            this.props.removeToPortfolio(value);
        }
    }

    renderSwitch(param) {
        switch(param) {
            case 0:
                return <WelcomeModal handleClose={this.handleClose} />;
            case 1:
                return <Details details={this.state.modalData} handleSwitch={this.handleSwitch} /> ;
            case 2:
                return <NewsModal handleClose={this.handleClose}/> ;
            case 3:
                return <ConfirmModal handleClose={this.handleClose}/> ;
            default:
                return <WelcomeModal handleClose={this.handleClose}/>;

        }
      }

    render() {
        if (!this.state.ready) {
            return (
                <ConnectionStatus colors={{ main: '#f26522' }} status={this.state.ready} error={this.state.errors} messages={this.state.messages} />
            )
        }
        else {
            return (
                <div>
                    {/* <Modal
                        open={this.state.modalDetails}
                        onClose={this.closeDetails}
                        aria-labelledby="simple-modal-title"
                        aria-describedby="simple-modal-description" >
                        <Details details={this.state.modalData} handleSwitch={this.handleSwitch} />
                    </Modal>*/}
                    <Modal
                        aria-labelledby="simple-modal-title"
                        aria-describedby="simple-modal-description"
                        open={this.state.modalToggle}
                        onClose={this.handleClose}>
                            {this.renderSwitch(this.state.typeContent)}
                    </Modal> 
                    <div className='masterContainer'>
                        <div className='barContainer'><MenuAppBar title="PERSONALIZE SEUS SERVIÇOS" /></div>
                        <div className='tabContainer'><TabContainer /></div>
                        <div className='cardsConteiner'><CardsProducts handleClose={this.handleClose} openDetails={this.openDetails}/></div>
                        <div className='footerContainer'><Footer /></div>
                    </div>
                </div>
            )
        }
    }
}

const mapStateToProps = state => ({
    online: state.auth.online,
    token: state.auth.token,
    msisdn: state.auth.msisdn,
    assinantesID: state.user.assinantesID,
    pontos: state.user.pontos,
    renovar: state.user.renovar,
    sva_produtos_id: state.user.sva_produtos_id,
    user_products: state.user.user_products,
    products: state.portfolio.products,
})

const mapDispatchToProps = dispatch => bindActionCreators({
    startConnection,
    signIn,
    getProducts,
    addToPortfolio,
    removeToPortfolio,
}, dispatch)

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(Home)