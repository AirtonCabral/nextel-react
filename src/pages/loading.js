import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import withStyles from '@material-ui/core/styles/withStyles';

const styles = theme => ({
  
});

export class Loading extends React.Component {

    constructor(props) {
        super(props);
        this.state = { };
    }
    
    render() {
      const { classes } = this.props;
      return (
        <div>
            {/* {'Loading Component'} */}
            {this.props.status !== '' &&
            <div>
                {this.props.status}
            </div>}
        </div>
      )
    }
}

Loading.propTypes = {
  classes: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
    pageLoaded: state.dom.page,
})

const mapDispatchToProps = dispatch => bindActionCreators({
    
}, dispatch)

export default connect(
    mapStateToProps,
    mapDispatchToProps,
)(withStyles(styles)(Loading))

