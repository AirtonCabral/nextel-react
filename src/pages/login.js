import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { loadPage } from '../actions/a_dom'
import { startConnection } from '../actions/a_auth'
import { getProducts } from '../actions/a_products'
import { signIn } from '../actions/a_user'
import CssBaseline from '@material-ui/core/CssBaseline';
import CircularProgress from '@material-ui/core/CircularProgress';
import orange from '@material-ui/core/colors/orange';
import withStyles from '@material-ui/core/styles/withStyles';
import { clearState } from './../lib/storage';

const styles = theme => ({
  layout: {
    width: 'auto',
    display: 'block', // Fix IE11 issue.
    marginLeft: theme.spacing.unit * 3,
    marginRight: theme.spacing.unit * 3,
    [theme.breakpoints.up(400 + theme.spacing.unit * 3 * 2)]: {
      width: 400,
      marginLeft: 'auto',
      marginRight: 'auto',
    },
  },
  paper: {
    marginTop: theme.spacing.unit * 8,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    padding: `${theme.spacing.unit * 2}px ${theme.spacing.unit * 3}px ${theme.spacing.unit * 3}px`,
  },
  avatar: {
    margin: theme.spacing.unit,
    backgroundColor: '#FF5722',
  },
  form: {
    width: '100%', // Fix IE11 issue.
    marginTop: theme.spacing.unit,
  },
  submit: {
    marginTop: theme.spacing.unit * 3,
    backgroundColor: '#FF5722'
  },
});


//////////////////////////
//  BASE NAME ////////////
//////////////////////////
const basename_home = 'home';
//////////////////////////

export class Login extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      login: '',
      password: '',
      isProcessing: false,
      buttonColorState: 'primary',
      buttonValueState: 'Entrar',
    };
    // console.log('-->> login builded');
  }

  loginApplication() {

    clearState();

    // let { login, password } = this.state;
    let msisdn = this.props.params.msisdn


    this.setState({ isProcessing: true, buttonValueState: 'Iniciando conexão' });

    // this.props.startConnection('drweb', 'n)CJL^?r4p#rYaG/R8A_', msisdn).then(() => { // login producao
    this.props.startConnection('drweb', 'c62J3rZovtw', msisdn).then(() => { // login dev homolog

      let buttonColorResult = this.props.online ? 'secondary' : this.state.buttonColorState;
      let buttonValueResult = this.props.online ? 'Seja bem vindo, Carregando Produtos...' : errorResultMessage;

      this.setState({
        buttonColorState: buttonColorResult,
        buttonValueState: buttonValueResult
      });

      this.props.getProducts().then(() => {
        this.setState({ buttonValueState: 'Baixando dados do Usuário...' });

        this.props.signIn().then(() => {

          // START PROJECT
          if (this.props.assinantesID !== null && this.props.assinantesID !== undefined) {
            setTimeout(() => {
              this.props.loadPage(basename_home)
            }, 300);
            this.setState({ buttonValueState: 'Tudo ok, iniciando...' });
          }
          else {
            this.updateStateOut('Ups, MSISDN Inválido.');
          }

        }) // end success signIn()
          .catch((error) => {
            this.updateStateOut('Ups, usuário indisponível ou instável. Recarregue a tela.');
          });

      }) // end success getProducts()
        .catch((error) => {
          this.updateStateOut('Ups, produtos insdisponíveis para este número "msisdn".');
        });

    }) // end success startConnection()
      .catch((error) => {
        this.updateStateOut('Ups, erro ao tentar se conectar. Recarregue a tela.');
      });

  }

  updateStateOut(msg) {
    this.setState({
      isProcessing: false,
      buttonValueState: msg
    });
  }

  componentDidMount() {
    // console.log('-->> login mounted');
    return this.loginApplication();
  }

  render() {

    const { classes, params } = this.props;
    const isCentral = 'atendente' in params && params.atendente !== '' ? true : false;
    const loginPageTitle = isCentral ? 'Central Serviços VAS' : 'Nextel Serviços VAS'
    const messageOut = this.state.buttonValueState;

    // const errorResultMessage = 'Error  :(  Recarregue a página.';

    // if (this.props.assinantesID !== null && this.props.assinantesID !== undefined) {
    //   messageOut = 'Logado!'
    // }

    if (this.props.userError !== '') {
      return (<div>
        {this.state.buttonValueState}
      </div>)
    }
    const stageHeight = window.innerHeight
    return (
      <div style={{display: 'flex', justifyContent: 'center', alignItems: 'center', height: stageHeight}}>
        <CircularProgress style={{ color: orange[600] }} />
      </div>
    )
  }
}

Login.propTypes = {
  classes: PropTypes.object.isRequired,
};

const mapStateToProps = state => ({
  online: state.auth.online,
  assinantesID: state.user.assinantesID,
  pageLoaded: state.dom.page,
  params: state.dom.params,
  userError: state.user.error,
})

const mapDispatchToProps = dispatch => bindActionCreators({
  startConnection,
  signIn,
  getProducts,
  loadPage,
}, dispatch)

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(withStyles(styles)(Login))

