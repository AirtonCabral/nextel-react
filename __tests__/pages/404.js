import React from 'react';
import shallow from 'react-test-renderer/shallow';
import { NotFound } from '../../src/pages/404'


// SÃ³ quero gerar o snapshot da primeira camada react.
const renderer = new shallow()

describe('Pagina de 404', () => {
    it('Renderiza corretamente', () => {
        const tree = renderer.render(<NotFound />)
        expect(tree).toMatchSnapshot()
    })
})