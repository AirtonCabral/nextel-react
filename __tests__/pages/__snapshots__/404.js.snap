// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagina de 404 Renderiza corretamente 1`] = `
<div
  style={
    Object {
      "paddingLeft": 50,
    }
  }
>
  <h1>
    PÃ¡gina nÃ£o encontrada :(
  </h1>
  <h2>
    <span
      role="img"
    >
      ðŸš‘ 
    </span>
    Volte para o seu 
    <Connect(ConnectedLink)
      to="/dashboard"
    >
      dashboard
    </Connect(ConnectedLink)>
  </h2>
</div>
`;
